### PR TITLE
Generex Build script

### PR DESCRIPTION
**Pull request by dthadi3**

### CODE REVIEW

**Comment by dthadi3** [comment]
[INFO] Installing /Generex/target/generex-1.0.2.jar to /root/.m2/repository/com/github/mifmif/generex/1.0.2/generex-1.0.2.jar
[INFO] Installing /Generex/pom.xml to /root/.m2/repository/com/github/mifmif/generex/1.0.2/generex-1.0.2.pom
[INFO] Installing /Generex/target/generex-1.0.2-javadoc.jar to /root/.m2/repository/com/github/mifmif/generex/1.0.2/generex-1.0.2-javadoc.jar
[INFO] Installing /Generex/target/generex-1.0.2-sources.jar to /root/.m2/repository/com/github/mifmif/generex/1.0.2/generex-1.0.2-sources.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 48.677 s
[INFO] Finished at: 2021-05-31T13:07:02+00:00
[INFO] Final Memory: 54M/415M
[INFO] ------------------------------------------------------------------------
[INFO] Scanning for projects...
[WARNING]
[WARNING] Some problems were encountered while building the effective model for com.github.mifmif:generex:jar:1.0.2
[WARNING] 'build.plugins.plugin.(groupId:artifactId)' must be unique but found duplicate declaration of plugin org.apache.maven.plugins:maven-javadoc-plugin @ line 212, column 12
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ line 226, column 12
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
[INFO] Inspecting build with total of 1 modules...
[INFO] Not installing Nexus Staging features:
[INFO]  * Preexisting staging related goal bindings found in 1 modules.
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building Generex 1.0.2
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-maven) @ generex ---
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ generex ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 0 resource
[INFO] skip non existing resourceDirectory /Generex/target/dependency
[INFO] skip non existing resourceDirectory /Generex/src/main/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ generex ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ generex ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Generex/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.5.1:testCompile (default-testCompile) @ generex ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-surefire-plugin:2.17:test (default-test) @ generex ---
[INFO] Surefire report directory: /Generex/target/surefire-reports
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-junit4/2.17/surefire-junit4-2.17.pom
[INFO] Downloaded: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-junit4/2.17/surefire-junit4-2.17.pom (3 KB at 3.5 KB/sec)
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-providers/2.17/surefire-providers-2.17.pom
[INFO] Downloaded: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-providers/2.17/surefire-providers-2.17.pom (3 KB at 52.8 KB/sec)
[INFO] Downloading: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-junit4/2.17/surefire-junit4-2.17.jar
[INFO] Downloaded: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-junit4/2.17/surefire-junit4-2.17.jar (63 KB at 826.5 KB/sec)

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.mifmif.common.regex.GenerexIteratorTest
Tests run: 20, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.862 sec - in com.mifmif.common.regex.GenerexIteratorTest
Running com.mifmif.common.regex.GenerexRandomTest
Tests run: 24, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.068 sec - in com.mifmif.common.regex.GenerexRandomTest
Running com.mifmif.common.regex.GenerexIteratorUnitTest
Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.161 sec - in com.mifmif.common.regex.GenerexIteratorUnitTest
Running com.mifmif.common.regex.GenerexProvidedRandomTest
Tests run: 24, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.051 sec - in com.mifmif.common.regex.GenerexProvidedRandomTest
Running com.mifmif.common.regex.GenerexUnitTest
Tests run: 16, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.513 sec - in com.mifmif.common.regex.GenerexUnitTest
Running com.mifmif.common.regex.GenerexTest
Tests run: 25, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.213 sec - in com.mifmif.common.regex.GenerexTest

Results :

Tests run: 121, Failures: 0, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 6.758 s
[INFO] Finished at: 2021-05-31T13:07:12+00:00
[INFO] Final Memory: 22M/233M
[INFO] ------------------------------------------------------------------------

File: g/Generex/Generex.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
```
Comment (gururajrkatti) [inline g/Generex/Generex.sh:1]: Pl. add the header comments. Refer build-scripts repos for any of the scripts to get the standard comments.

**Review by gururajrkatti — CHANGES_REQUESTED** [review]

File: g/Generex/Generex.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+
+apt install  -y maven
+
+mvn -v
+
+
+git clone ${REPO}
+cd Generex
+git checkout ${VERSION}
```
Comment (gururajrkatti) [inline g/Generex/Generex.sh:52]: what if VERSION wrong/git checkout fails ?

**Review by gururajrkatti — CHANGES_REQUESTED** [review]

File: g/Generex/Generex.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+mvn clean install -DskipTests=true -B -V
```
Comment (gururajrkatti) [inline g/Generex/Generex.sh:53]: If "mvn clean install" fails, should we continue with "mvn test" still ?

**Review by gururajrkatti — CHANGES_REQUESTED** [review]

**Comment by dthadi3** [comment]
root@120ce1cb2f4d:/Generex# cd target/
root@120ce1cb2f4d:/Generex/target# ls
apidocs                 classes              generated-classes          java-imports.xml         maven-status
checkstyle-cachefile    cobertura            generex-1.0.2-javadoc.jar  java-unusedcode.xml      pmd.xml
checkstyle-checker.xml  cpd.xml              generex-1.0.2-sources.jar  javadoc-bundle-options   site
checkstyle-header.txt   findBugsExclude.xml  generex-1.0.2.jar          javancss-raw-report.xml  surefire-reports
checkstyle-result.xml   findbugsXml.xml      java-basic.xml             maven-archiver           test-classes

**Comment by dthadi3** [comment]
@gururajrkatti please check now.

**Review by gururajrkatti — APPROVED** [review]
ok

**Pull request merged**